docs/architecture-decisions/adr002-default-catalog-file-format.md

```diff
@@ -0,0 +1,245 @@
+          we will focus on the <code>Component</code> kind.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>metadata</code></td>
+      <td valign="top">Object</td>
+      <td valign="top">Yes</td>
+      <td valign="top">
+        <p>
+          Metadata about the entity, such as its name, that are not directly part of
+          the entity specification itself. See below for more details about this structure.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>spec</code></td>
+      <td valign="top">Object</td>
+      <td valign="top">No</td>
+      <td valign="top">
+        <p>
+          The actual specification data that describes the entity.
+        </p>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+The `metadata` root field has the following nested structure.
+
+<table>
+  <thead>
+    <tr>
+      <th>Field</th>
+      <th>Type</th>
+      <th>Required</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td valign="top"><code>name</code></td>
+      <td valign="top">String</td>
+      <td valign="top">Yes</td>
+      <td valign="top">
+        <p>
+          The name of the entity. Names must be unique per kind, within a given namespace.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>namespace</code></td>
+      <td valign="top">String</td>
+      <td valign="top">No</td>
+      <td valign="top">
+        <p>
+          The `name` of a namespace that the entity belongs to. This field is optional.
+        </p>
+        <p>
+          Namespaces may also be part of the catalog, and are <code>v1</code>/<code>Namespace</code>
+          objects, i.e. not Backstage specific but the same as in Kubernetes.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>description</code></td>
+      <td valign="top">String</td>
+      <td valign="top">No</td>
+      <td valign="top">
+        <p>
+          A human readable description of the entity, to be shown in Backstage. May contain
+          markdown.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>labels</code></td>
+      <td valign="top">Object</td>
+      <td valign="top">No</td>
+      <td valign="top">
+        <p>
+          Labels are key/value pairs of that are attached to the entity, and their use is
+          identical to <a href="https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/">kubernetes object labels</a>.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>annotations</code></td>
+      <td valign="top">Object</td>
+      <td valign="top">No</td>
+      <td valign="top">
+        <p>
+          An object with arbitrary non-identifying metadata attached to the entity,
+          identical in use to <a href="https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/">kubernetes object annotations</a>.
```

Comment:
Who will be consuming this data? I'm thinking this is backend-only info that for example could be used as hints to an inventory aggregator or smth like that. I don't think we should add them before we have a use-case for them anyway.

Reply:
The most obvious use case here as far as I can see is git info added at ingestion time. Org, repo, path, branch name, ref spec / hash.

However, this is like THE way to grow your model, as long as we haven't yet invented a full blown "sub-fact" system. For example, where you would specify refs to pagerduty. In fact, this key space is very similar to what you suggested elsewhere; I could envision having some well known keys here, such as

```yaml
metadata:
  annotations:
    pagerduty.com/oncall-schedule: "..."
```

and then the user interface reacts to this, through enabling view features based on the presence or absence of annotations. Same thing for Git revision info - we could link out to the repo etc, based on whether this info is present or not.

I think this may turn out to be a very important feature, if used right.

Reply:
👍 This is where we could put keys into external systems. Other similar cases would be K8S cluster/namespace/service names for backend services (enabling the Kubernetes view), Spinnaker/CircleCI/Jenkins pipelines (enabling CI view), etc.

Reply:
Yeah I would be happy with this being THE fact system tbh. The only thing I'm somewhat missing is the lack bundling of things, like being able to do:

```yaml
annotations:
  github.com/repo:
    org: spotify
    repo: backstage
```

I know ^ is a bit silly, but it's just an example :grin:

Reply:
The purpose of the bundles would be the possibility to do some additional validation, and to have tightly coupled values be a bit more human readable.

A better set of examples:

### A

```yaml
annotations:
  spotify.com/tugboat/notifications/method: slack
  spotify.com/tugboat/notifications/target: tools-private
  spotify.com/tugboat/notifications/types: deploymentFailed,discrepancy
```

### B

```yaml
annotations:
  spotify.com/tugboat/notifications: method=slack;target=tools-private;types=deploymentFailed,discrepancy
```

### C

```yaml
annotations:
  spotify.com/tugboat/notifications:
    method: slack
    target: tools-private
    types: [deploymentFailed, discrepancy]
```

Do you prefer A, B, C, or D? :grin:

Reply:
Annotations are free form, so you would be perfectly able to do C (or make it actually a list of such objects if you want to have more than one notification type)

Reply:
In k8s they're not though, so we 'd be diverging there, and it would be a bit unfortunate to not be able to reuse tooling support for k8s manifests just because of this change

Reply:
I think that's incorrect. The kubernetes documentation says:

`The metadata in an annotation can be small or large, structured or unstructured, and can include characters not permitted by labels.`

https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/

Reply:
It also intentionally mentions no format restrictions on the value in that document, while the corresponding document for labels does list the value restrictions very clearly.

Reply:
This is what you get if you try it with kubectl:

```
error validating data: ValidationError(Deployment.metadata.annotations.herp-derp.com/wutter): invalid type for io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta.annotations: got "map", expected "string"
```

Reply:
You can ofc still do this though:

```yml
annotations:
  spotify.com/tugboat/notifications: |
    method: slack
    target: tools-private
    types: [deploymentFailed, discrepancy]
```

<img src="https://i.imgur.com/dcc6Gsmg.jpg" width="300"/>

Reply:
fixed